### ISSUE TITLE
Unexpected DX12 Device Removal on Fence Creation

### ISSUE THREAD

**Issue by rnd-ash** [gfx-rs/wgpu#1382]
Raising this as part of [this issue](https://github.com/rnd-ash/OpenVehicleDiag/issues/25), for [OpenVehicleDiag](https://github.com/rnd-ash/OpenVehicleDiag)

**Description**
Using the Iced GUI Library, it appears WGPU panics with the DX12 backend with the following stack trace:
```
thread 'main' panicked at 'assertion failed: (left == right)
left: 0,
right: -2005270523', C:\Users\Ashcon.cargo\registry\src\github.com-1ecc6299db9ec823\gfx-backend-dx12-0.7.0\src\device.rs:1032:9
stack backtrace:
0: 0xc986c8 -
```

**Expected vs observed behavior**
Program does not crash

**Comment by cwfitzgerald** [comment]
Alright, we appear to be hitting a device removal (error code is 0x887A0005 DXGI_ERROR_DEVICE_REMOVED) some time before we call CreateFence, then the fence creation is exposing this removal. In order to diagnose this, we need to find the reason for device removal, which will involve running the program in the MSVC debugger to get the error message from the debug layers.

This is for an old version of wgpu/gfx-backend-dx12 (current is 0.8) because iced [hasn't updated yet](https://github.com/hecrj/iced/pull/830), but there is a patch update available in the 0.7 train for gfx-backend-dx12, so you should try `cargo update gfx-backend-dx12` to get the latest patch and try with that. 

---

I should note that we should also hook into this debug information like we do the vulkan validation layers.

**Comment by cwfitzgerald** [comment]
Closing as stale due to hal rewrite, please refile if it's still an issue.
